### PR TITLE
Implementing a Generic Singly linked list Data Structure in Java

### LINKED ISSUE #4382: Implementing a Generic, singly Linked List data structure in Java
#### Proposed title of article
Implementing a Generic, singly Linked List data structure in Java
#### Proposed article introduction
Linear Data Structures that store values in nodes are known as Linked Lists. As we know, each Node has two properties: the value of the node and, if present, a connection to the next node. Linked Lists can be of any data type, including String, Boolean, Float, Character, and so on. 
We will be creating a generic Linked List Data Type that can store data of any type.

#### Key takeaways 
- Primary member functions of a linked list
- Creating a generic linked list
- What is a generic linked list and how we implement them

#### References
N/A
### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by LynnMumo**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #4382

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Kayere** [comment]
Plagiarism check done ✔️

File: content/articles/implementing-a-generic-singly-linked-list-data-structure-in-java/index.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+#### Primary linked list member functions
+These are functions that can be performed on a linked list:
+- `Sort` - Sorts the linked list's nodes.
+- `Search` - Chooses one of the related nodes and clicks on it.
+- `Deletion` - Gets rid of everything that's already there.
+- `Insertion` - Extends the linked list by one element.
+- `Traversal` - Accesses the connected list's elements one by one.
+
+### Implementation of a Java single-linked list
+In this section, we'll explore how generics can be used to implement a singly linked list in a type-safe, parameterized fashion. We will use the Java program below to generate our own type-safe linked list in the Java programming language. It is available for download here.
```
Comment (Dawe-7) [inline content/articles/implementing-a-generic-singly-linked-list-data-structure-in-java/index.md:50]: The last sentence here, what is available for download?

File: content/authors/lynn-mumo/index.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+title: Lynn Mumo
+type: authors
+github: https://github.com/LynnMumo
+---
```
Comment (Dawe-7) [inline content/authors/lynn-mumo/index.md:5]:
```suggestion
github: https://github.com/LynnMumo
  - url: /engineering-education/authors/lynn-mumo/avatar.jpg
---
```
Add this after your github link

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
@LynnMumo, welcome to Section's EngEd and congratulations on submitting your first article. The article is an interesting read and to make it better I've left you a few suggestions. Also, your `avatar.jpg` should be a snapshot of yourself and not a random image.

Ping me when done so we can move on to tech review. Thanks.

**Comment by LynnMumo** [comment]
@Dawe-7 Done with the edits. Thanks.

**Comment by Dawe-7** [comment]
@LynnMumo, kindly add the snapshot of yourself in your 'avatar.jpg' file. You missed on that  .

**Comment by LynnMumo** [comment]
> @LynnMumo, kindly add the snapshot of yourself in your 'avatar.jpg' file. You missed that.

Adding it right away @Dawe-7

**Comment by LynnMumo** [comment]
@Dawe-7 Done

**Review by Dawe-7 — COMMENTED** [review]
@LynnMumo, upon running your code, i got errors. Kindly make sure your code runs. [Here](https://replit.com/@Dawe7/codeTest2#Main.java) is a link of the code test. After you've made sure that your code runs, add [replit](https://replit.com/~) links in your article before the output to make it easier for the reader to run your code.

Use this template after the code:
```
Go ahead and run the code [here](replit link provided here)
```

**Comment by LynnMumo** [comment]
@Dawe-7 Done

**Comment by Dawe-7** [comment]
@LynnMumo, your code works, but, a brief explanation of what is happening in the code would be very helpful to readers. Also, could we have more comments in the code.  Plus, do the same thing you've done with the first code, add the code output and the quick-run link. 

Do this and your tutorial will be ready for final review.

**Comment by LynnMumo** [comment]
> @LynnMumo, your code works, but, a brief explanation of what is happening in the code would be very helpful to readers. Also, could we have more comments in the code. Plus, do the same thing you've done with the first code, add the code output and the quick-run link.
> 
> Do this and your tutorial will be ready for final review.

Done @Dawe-7

**Review by Dawe-7 — APPROVED** [review]
Great first article @LynnMumo. Kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles. Thank you, looking forward to more of your contributions in the EngEd program.

@Kayere, this article is ready for final review.

**Comment by lalith1403** [comment]
Jumping on this one

File: content/articles/implementing-a-generic-singly-linked-list-data-structure-in-java/index.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+images:
+
+  - url: /engineering-education/implementing-a-generic-singly-linked-list-data-structure-in-java/hero.jpg
+    alt: Implementing a Generic Singly linked list Data Structure in Java Hero Image
+---
+A `linked list` is used to hold node-based data. Data in a linked list is spread across many locations rather than being stored in a single memory location like in an array. 
+<!--more-->
+Each node has a reference section where an address of the next node is maintained. The brief linked list has links connecting some of the nodes. 
+
+By removing the challenges of an array, such as the need for contiguous memory and the difficulty of array insertion and deletion, an array data structure is improved. A linked list, but, makes it simple to add and delete items as your requirements change.
```
Comment (lalith1403) [inline content/articles/implementing-a-generic-singly-linked-list-data-structure-in-java/index.md:21]: Could not understand this sentence. Can you work on it?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Make the change and let me know.

**Comment by lalith1403** [comment]
Additionally, do a thorough check once. There were several errors that I noticed at a glance. 

@Dawe-7 Please check for any errors once the edits are made. Thank you :)

**Comment by LynnMumo** [comment]
> Make the change and let me know.

Done @lalith1403

**Comment by Dawe-7** [comment]
@LynnMumo, ping me when done with the changes.

**Comment by LynnMumo** [comment]
@Dawe-7 already done. Thanks

**Review by Dawe-7 — APPROVED** [review]
Thanks for the changes @LynnMumo 

@lalith1403, peer review has been completed. Could we have a final review, thanks.

**Comment by lalith1403** [comment]
The article is an introduction to linked list in Java. You should talk about the code that explains the logic behind building and using linked lists. Moreover, the explanations are not clear. Work on it and note the comments made. 

It is very importatnt we deliver absolute value that is scarce on the web. If we are adding value in terms of content that is easily avaialble, we make sure to take it a step further and break down the details of implementing the concepts.

**Comment by lalith1403** [comment]
This PR needs a lot of revision. I will be closing this PR. Please feel free to submit a revised and edited new PR. Ping us once done. Thank you.

**Pull request closed without merging**